python/ray/serve/_private/deployment_state.py

```diff
@@ -1368,34 +1407,35 @@ def _set_target_state(self, target_info: DeploymentInfo) -> None:
             ):
                 ServeUsageTag.NUM_REPLICAS_LIGHTWEIGHT_UPDATED.record("True")
 
-        self._target_state = target_state
-        self._curr_status_info = DeploymentStatusInfo(
-            self.deployment_name, DeploymentStatus.UPDATING
-        )
-        self._replica_constructor_retry_counter = 0
-        self._backoff_time_s = 1
```

Comment:
Why don't we reset these values inside `_set_target_state()` anymore? Resetting them here is nice because it's easy to forget to reset them explicitly after calling `_set_target_state()`.

Reply:
`_set_target_state` is only called in autoscale() and deploy(). We don't want to reset these values in `autoscale()` so I moved it to deploy()